lib/cache/redis.go

```diff
@@ -194,14 +194,20 @@ func (r *Redis) Get(key string) ([]byte, error) {
 	return []byte(res), nil
 }
 
-// Set attempts to set the value of a key.
-func (r *Redis) Set(key string, value []byte) error {
+// SetWithTTL attempts to set the value of a key.
+func (r *Redis) SetWithTTL(key string, value []byte, ttl *time.Duration) error {
 	r.mSetCount.Incr(1)
 	tStarted := time.Now()
 
 	key = r.prefix + key
 
-	err := r.client.Set(key, value, r.ttl).Err()
+	var t time.Duration
+	if ttl != nil {
+		t = *ttl
+	} else {
+		t = r.ttl
+	}
+	err := r.client.Set(key, value, t).Err()
```

Comment:
The next call within the retry loop needs the provided ttl as well.

Reply:
Done